R/render-table.R

```diff
@@ -37,23 +104,96 @@ renderTable <- function(expr, ..., env=parent.frame(), quoted=FALSE, func=NULL)
     xtable_args <- dots[intersect(names(dots), xtable_argnames)]
     non_xtable_args <- dots[setdiff(names(dots), xtable_argnames)]
 
-    # Call xtable with its args
+    # A small helper function to determine if the row.names can be coerced
+    # to numeric or if they are legitimate strings
+    isNumber <- function(n) {
+      grepl("^\\d$+", n)
```

Comment:
It would be better to check the underlying `row.names()` attribute:

```R
df1 <- data.frame(x = 1:2)
is.numeric(attr(df1, "row.names"))
df2 <- data.frame(x = c(a = 1, b = 2))
is.numeric(attr(df2, "row.names"))
```